multicluster/controllers/multicluster/leader/member_cleanup_controller.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2023 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leader
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+)
+
+// MemberResourcesCleanupController will remove all ResourceExports belongs to a member
+// cluster when the corresponding MemberClusterAnnounce CR is deleted. It will also try
+// to clean up all stale ResourceExports during start.
```

Comment:
It is not clear to me how we divide functions between MemberResourcesCleanupController and SlateController. Could you clarify?

Reply:
For a leader cluster:
`MemberResourcesCleanupController` is mainly to check the deletion event of MemberClusterAnnounce to make sure all corresponding ResourceExports will be removed in the leader cluster for the deleted member cluster.
`SlateController` is responsible to watch the `MemberClusterAnnounce`'s timestamp annotation and delete it if it is not updated over 24 hours.